### PR TITLE
[Relay][Frontend][TFlite] Add support for quantized LOGISTIC

### PR DESCRIPTION
**Pull request by inadob**
This patch is built on top of two pending PRs: #4789 and #4805 (only 70dfaf6) 
 * add qnn implementation in the parser
 * add test case for qnn logistic

### CODE REVIEW

**Comment by inadob** [comment]
@anijain2305 @FrozenGene, can you please check this patch

**Comment by u99127** [comment]
> 
> 
>     * add qnn implementation in the parser
> 
>     * add test case for qnn logistic

LGTM

Ramana

**Comment by u99127** [comment]
ping for an approving review ? @FrozenGene @anijain2305.

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -384,7 +401,16 @@ def convert_logistic(self, op):
         input_tensor = input_tensors[0]
         in_expr = self.get_expr(input_tensor.tensor_idx)
 
+        output_tensors = self.get_output_tensors(op)
+        assert len(output_tensors) == 1, "output tensors length should be 1"
+        output_tensor = output_tensors[0]
+
+        if input_tensor.qnn_params:
+            in_expr = self.dequantize(in_expr, input_tensor)
         out = _op.sigmoid(in_expr)
```
Comment (FrozenGene) [inline python/tvm/relay/frontend/tflite.py]: Does the INT computation of sigmoid is very complex so that we could only `dequantize->float compute->requantize`?

Comment (inadob) [reply python/tvm/relay/frontend/tflite.py]: Well I have the following understanding for that case: since by definition sigmoid outputs a probability in [0, 1] in real space, then in uint8 [0, 1] will have a different meaning, so we should dequantize before doing the operation. Correct me if I'm wrong.
Or you are talking about INT, not UINT?

Comment (FrozenGene) [reply python/tvm/relay/frontend/tflite.py]: I was talking about **pure integer computation of sigmoid.** That is to say, current computation of quantized sigmoid is float arithmetic in fact, we just `dequantize` to float, `float arithmetic`, `requantize` to quantized UINT8 type. However, like TFLite, we have **pure integer computation of sigmoid.** like this : https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/internal/reference/integer_ops/logistic.h#L24, which is I checked just now. It seems a little complicated. So maybe we use current way is an acceptable way.

Comment (u99127) [reply python/tvm/relay/frontend/tflite.py]: If there is a need to improve quantized logistic performance, introducing a qnn.logistic sounds reasonable and then lowering it as part of the legalize_qnn pass would be what I would do: However that in my mind is the next step in terms of improving the performance for logistic , the first step on the other hand to make this work and the dequantize , operation, quantize is a good way of making progress.

Comment (anijain2305) [reply python/tvm/relay/frontend/tflite.py]: I had exact similar thoughts. As @FrozenGene pointed, the implementation for integer-only softmax looked complicated, so I decided to just support functionally for now.

File: tests/python/frontend/tflite/test_forward.py [hunk 2]
```diff
@@ -143,11 +144,13 @@ def compare_tflite_with_tvm(in_data, in_name, input_tensors,
...
             input_arrays = converter.get_input_arrays()
             input_stats = {}
-            # hardcode the mean_values and std_dev_values (m,s) to be the same
-            # if all inputs are in (float_min; float_max) == (-100, 100)
+            # calculate the mean and quantization scale for every input tensor,
+            # with respect to its fp32 input range, defined in fake_quant.
             # s = 255/(fmax-fmin);  m = -fmin*s (the zero point)
             for i in input_arrays:
-                input_stats[i] = (128., 1.275)
+                quant_scale = 255 / (input_range[i][1] - input_range[i][0])
```
Comment (anijain2305) [inline tests/python/frontend/tflite/test_forward.py]: Very Minor - a check to ensure denominator is not zero

Comment (inadob) [reply tests/python/frontend/tflite/test_forward.py]: I updated this here #4789. Once we agree on the changes in that PR, I will update all the others depending on it.

**Comment by anijain2305** [comment]
@inadob Please rebase

**Comment by tqchen** [comment]
@inadob please rebase, @FrozenGene please https://docs.tvm.ai/contribute/code_review.html#approve-and-request-changes-explicitly

**Comment by FrozenGene** [comment]
Thanks @inadob @anijain2305 @u99127 This is merged.
